lib/active_record/virtual_attributes/virtual_total.rb

```diff
@@ -85,10 +106,10 @@ def define_virtual_size_method(name, relation)
 
       def define_virtual_aggregate_method(name, relation, method_name, column)
         define_method(name) do
-          if attribute_present?(name)
-            self[name]
+          if has_attribute?(name)
+            self[name] || 0
           elsif (rel = send(relation)).loaded?
-            rel.blank? ? nil : rel.map { |t| t.send(column) }.compact.send(method_name)
+            rel.map { |t| t.send(column) }.compact.send(method_name)
```

Comment:
yea, this looks just like the changes in 15.

This method seems to handle all our use cases. No reason to define individual methods